api/src/database/repositories/ReplyRepository.ts

```diff
@@ -15,11 +15,11 @@ export const ReplyRepository = AppDataSource.getRepository(Reply).extend({
       .getMany();
   },
 
-  findReplyWithPropertiesById(replyId: number) {
+  findReplyWithPropertiesById(replyId: number): Promise<Reply> {
     return this.createQueryBuilder('reply')
       .leftJoinAndSelect('reply.user', 'user')
       .leftJoinAndSelect('reply.snippets', 'snippet')
       .where('reply.id = :replyId', { replyId })
-      .getOne();
+      .getOneOrFail();
```

Comment:
Q: how does the server behaves when this fails? Like does it shut down the server or throw out some sort of error message with the server running?

Reply:
Local development server will crash and send an error response. Production will handle it and send the error without crashing. Will probably store it in logs as well, I have not tested yet.

Reply:
as long as it doesn't break the server I'm fine.

Reply:
@shawonibnkamal ^^

Reply:
> as long as it doesn't break the server I'm fine. Sidenote: it it breaks in the local then it would become harder to test as we can't test before releasing to prod and if sth goes wrong in the prod we are doomed. So maybe better option would be to handle the failure in a diff way? thoughts?

Not sure what you mean. If it breaks locally, it means we have a deeper issue and we should fix it. If it don't break locally then there is nth to worry.

Reply:
> Local development server will crash and send an error response.

If the local server crashes how are we planning to test it locally. also isn't it too risky to push it to prod without testing locally. So wondering if we can handle the error in a diff way.

Reply:
Local development uses nodemon. On any unhandled exception, we are logging it and closing the process. Nodemon cannot handle multiple processes, so it crashes.

On production, we are using pm2. On any unhandled exception, we are logging it and closing the process. But pm2 can handle multiple processes. So server do not crash.

Local server will crash only when a reply is not found. Which ideally should never happen. Because we should not send a replyId from the client that do not already exist.

We can have try-catch methods on errors we expect to be frequent. If it is unexpected, it is not worth doing try-catch.